### PR TITLE
MET-2935: Determine availability of depublication.

### PR DESCRIPTION
**Pull request by jochen-vermeulen**

### CODE REVIEW

File: metis-core/metis-core-rest/src/main/java/eu/europeana/metis/core/rest/DepublishRecordIdController.java [hunk 1]
```diff
@@ -165,7 +165,7 @@ public void deletePendingRecordIds(@RequestHeader("Authorization") String author
       MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE})
   @ResponseStatus(HttpStatus.OK)
   @ResponseBody
-  public ResponseListWrapper<DepublishRecordIdView> getDepublishRecordIds(
+  public DepublicationInfoView getDepublishRecordIds(
```
Comment (stzanakis) [inline metis-core/metis-core-rest/src/main/java/eu/europeana/metis/core/rest/DepublishRecordIdController.java:168]: Update javadoc return text

File: metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/DepublishRecordIdService.java [hunk 2]
```diff
@@ -276,4 +278,46 @@ public WorkflowExecution createAndAddInQueueDepublishWorkflowExecution(MetisUser
...
+   */
+  public boolean canTriggerDepublication(MetisUser metisUser, String datasetId)
+          throws GenericMetisException {
+
+    // Authorize.
+    authorizer.authorizeReadExistingDatasetById(metisUser, datasetId);
+
+    // If a workflow execution is currently in progress, we can't depublish.
+    if (orchestratorService.getRunningOrInQueueExecution(datasetId) != null) {
+      return false;
```
Comment (stzanakis) [inline metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/DepublishRecordIdService.java:314]: Combine the two return types into one, using an if/else block

File: metis-core/metis-core-rest/src/main/java/eu/europeana/metis/core/rest/DepublishRecordIdController.java (hunk 1 above)
Comment (jochen-vermeulen) [reply metis-core/metis-core-rest/src/main/java/eu/europeana/metis/core/rest/DepublishRecordIdController.java:168]: Done.

File: metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/DepublishRecordIdService.java (hunk 2 above)
Comment (jochen-vermeulen) [reply metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/service/DepublishRecordIdService.java]: Done.

**Pull request merged**